### DISCUSSION TITLE
ch1-basic/ch1-04-func-method-interface

### DISCUSSION THREAD

**Discussion by giscus** [chai2010/advanced-go-programming-book#605] [General]
# ch1-basic/ch1-04-func-method-interface

Go语言高级编程

https://chai2010.cn/advanced-go-programming-book/ch1-basic/ch1-04-func-method-interface.html

**Comment by sfx2zh1** [comment]
在Go 1.22之后 loopvar 变成每个循环都重新生成了 1.4.1节的闭包对loopvar的引用部分需要重新改了 https://go.dev/wiki/LoopvarExperiment
```
func main() {
    for i := 0; i < 3; i++ {
        defer func(){ println(i) } ()
    }
}
```
的output在go1.22之后是2 1 0

**Reply by chai2010** [reply]
Go1.22是错误的设计，是通关补丁来破坏规则。这个错误设计是从js继承而来

**Comment by 2019wangmeng** [comment]
1.4.1中第一句话，在 Go 语言中，函数是第一类对象，我们可以将函数保持到变量中。这里的保持是不是应该是保存，是不是字打错了，还是说我理解有误。

**Comment by changkaiyan** [comment]
您好，我将尽快回复您的邮件。如果您的邮件无需回复，则收到本回复表明我已收到您的邮件。请勿回复，谢谢合作。This is an automatic reply, confirming that your e-mail was received.Thank you.
